### ISSUE TITLE
When I call the .train( ) method, what algorithm is it being trained with?

### ISSUE THREAD

**Issue by ParthJawale1996** [flairNLP/flair#883]
When I see the .train() giving me something like this when I start running the code: 
Model: "TextClassifier(
  (document_embeddings): DocumentPoolEmbeddings(
    fine_tune_mode=linear, pooling=mean
    (embeddings): StackedEmbeddings(
      (list_embedding_0): WordEmbeddings('en-news')
    )
    (embedding_flex): Linear(in_features=300, out_features=300, bias=False)
  )
  (decoder): Linear(in_features=300, out_features=2591, bias=True)
  (loss_function): CrossEntropyLoss()

What does that tell me about the model on which is being used for training? 
@alanakbik @stefan-it

**Comment by alanakbik** [comment]
This is a printout of the PyTorch model that you are training, consisting of an embedding layer, a decoder and a loss function. All Flair models are also PyTorch models, so we use the same notation for printing them.

**Comment by ParthJawale1996** [comment]
@alanakbik  So basically this is running an RNN? What is the structure of the architecture? Like how many cells do we have here? And how do I tweak these models that are being used?
